Functions/Import/xASL_imp_DCM2NII_Subject_SortASLVolumes.m

```diff
@@ -165,19 +165,25 @@
 
 
 %% Determine if we have a Hadamard sequence based on the parameters of the studyPar.json
-function [bTimeEncoded] = xASL_imp_DCM2NII_CheckIfTimeEncoded(x, bTimeEncoded)
+function [bTimeEncoded] = xASL_imp_DCM2NII_CheckIfTimeEncoded(x, bTimeEncoded, iSubject, iVisit, iSession)
```

Comment:
Part one of your question - it is specific to Hadamard. We are figuring out if this is hadamard or isn't, because we need that information for Hadamard. But we don't need to know in the import if a sequence is multi-TE or multi-PLD. So that's why we need this function and don't need the others.

Part two - I don't understand that part of the question. It doesn't do an import overview. It is just trying to figure out if a specific subject+visit+session+run is or isn't hadamard, because then the import is done a bit differently.

Overall - this is all fine to me.

Reply:
It was defining general subject/visit/session fields, that you need in multiple import related functions, not only in the Hadamard function. See the structs below, these definition are needed throughout the import, not only for Hadamard.

>>	studyParAll = xASL_io_ReadDataPar(x.dir.studyPar, true);
	structSubject = x.importOverview.(['subject_' num2str(iSubject,'%.3d')]);
	structVisit   = structSubject.(['visit_' num2str(iVisit,'%.3d')]);
	structRun     = structVisit.(['run_' num2str(iSession,'%.3d')]);
	studyParSpecific = xASL_imp_StudyParPriority(studyParAll, structSubject.name, structVisit.name, structRun.regexp(2:end));

Reply:
It is a very special case where you need to know in the DCM2NII step if a scan is hadamard or not. you don't need this anywhere else.
We are not defining any fields, none of this leaves this function.

Reply:
We are going to need this somewhere else, since there will be more cases (e.g., is this velocity encoding yes/no, etc.). Hadamard won't be the only exception to PCASL that we will use. So I would move this to a more general parent function (in fact, I believe we usually already have something like `x.iSubject`, `x.SubjectName` etc.).

This would make the code more readable. The priority of study pars is a general thing, the reading of data par is general, etc. Not only for a minor sub-sub-function that checks if we have Hadamard or not.

But if I don't convince you with this, just resolve this conversation :)

Reply:
Yes, haven't convinced me :D

I mean - I fully agree. In a general case. But here, we need to know this only for a small specific subtask of DICOM volume sorting. So we can only hardly move this elsewhere, because we only need it here. We don't need to know these things in general for other sequences. So this is really a minor sub-sub-function that does this workaround for this difficult hadamard case.

Reply:
It is probably already defined somewhere else. And we WILL need it somewhere else, just wait until we start implementing velocity encoded and all other fancy flavors ;)

But OK for now. Please resolve this.